train-user/src/main/java/hu/bme/mit/train/user/TrainUserImpl.java

```diff
@@ -30,8 +34,21 @@ public int getJoystickPosition() {
 
 	@Override
 	public void overrideJoystickPosition(int joystickPosition) {
+
 		this.joystickPosition = joystickPosition;
 		controller.setJoystickPosition(joystickPosition);
-	}
 
+		if(!timer_exists)
+		{
+			Timer timer = new Timer();
```

Comment:
The timer variable should be a class variable, so we have a reference to it and we can cancel it later if its needed.